File: trunk/C#/Data Fairy/Data Fairy Editor/Code Templates/C#/AutoTests.cs
using PACKAGE_STRING.dbvos;
using PACKAGE_STRING.interfaces;

namespace PACKAGE_STRING
{
	public class AutoTests
	{
		public static void Main(string[] args)
		{
			IDBVOsModel model = new DBVOsModel();
			foreach(IDBVOTable table in model.tableList)
			{
				System.Console.WriteLine("Table: " + table.tableName + ", Rows: " + table.rowList.Count);
			}
		}
	}
}

File: trunk/C#/Data Fairy/Data Fairy Editor/Code Templates/PHP/AutoTests.php
<?php
function __autoload($class_name) {
	$file = $class_name . '.class.php';

	if(file_exists('base/' . $file)) {
		require_once('base/' . $file);
		return;
	}

	if(file_exists('dbvos/' . $file)) {
		require_once('dbvos/' . $file);
		return;
	}

	if(file_exists('enums/' . $file)) {
		require_once('enums/' . $file);
		return;
	}

	if(file_exists('interfaces/' . $file)) {
		require_once('interfaces/' . $file);
		return;
	}

	if(file_exists('tables/' . $file)) {
		require_once('tables/' . $file);
		return;
	}

	throw new Exception("Unable to find class file " . $file . " for type " . $class_name);
}

class AutoTests
{
	public function __construct()
	{
		$model = new DBVOsModel();
		foreach($model->tableList as $table)
		{
			print ("<p>Table: " . $table->tableName . ", Rows: " . count($table->rowList) . "</p>");
		}
	}		
}

$test = new AutoTests();